packages/dcos-integration-test/extra/test_metrics.py

```diff
@@ -32,48 +33,68 @@ def test_metrics_masters_ping(dcos_api_session):
         assert response.json()['ok'], 'Status code: {}, Content {}'.format(response.status_code, response.content)
 
 
+def telegraf_prometheus_request(host):
+    """Fetch metrics from Telegraf's Prometheus endpoint.
+    """
+    return requests.get(
+        url='http://' + host + ':61091/metrics',
```

Comment:
I would still like to understand if this is a private or a public interface.

From the look of it I think `dcos_api_session.metrics...` probably uses a _public_ interface.

Generally, in this test suite we should focus on testing _public_ interfaces. If this is a private interface we should have a code comment at the very least so that future maintainers know that it's maybe ok to break this/a test.

@philipnrmn can you comment on this?

Reply:
I don't understand what you mean by public and private interfaces. I guess you mean exposed by Admin Router (public) and exposed on `0.0.0.0` (private in the sense of the DC/OS internal network?). 
What I can say: Currently there is no way to query these metrics through Admin Router. Therefore the test needs to be written in this way.

Reply:
I'll leave it to @jgehrcke to explain precisely what he meant, but to help things along @timaa2k perhaps you can answer the following:

If we ignore this test, are DC/OS developers committed (e.g. via documentation) to keep the interface you have used here (`http://<host>:61091/metrics`) working as-is, or is the fact that this works an implementation detail for another "public" interface (something that is documented)?

Reply:
The 1.12 documentation state pretty clearly that metrics are (publicly) available on port `61091` on every node: https://docs.mesosphere.com/1.12/metrics/#overview.
According to Cluster Ops the `/metrics/v0/` API will be deprecated soon.